src/segment-loader.js

```diff
@@ -923,22 +925,53 @@ export default class SegmentLoader extends videojs.EventTarget {
     if (segment.key) {
       // this is an encrypted segment
       // incrementally decrypt the segment
-      /* eslint-disable no-new, handle-callback-err */
-      new Decrypter(segmentInfo.encryptedBytes,
-                    segment.key.bytes,
-                    segment.key.iv,
-                    (function(err, bytes) {
-                      // err always null
-                      segmentInfo.bytes = bytes;
-                      this.handleSegment_();
-                    }).bind(this));
-      /* eslint-enable */
+      this.decrypter_.postMessage({
+        action: 'decrypt',
+        source: this.loaderType_,
+        encrypted: {
+          bytes: segmentInfo.encryptedBytes.buffer,
+          byteOffset: segmentInfo.encryptedBytes.byteOffset,
+          byteLength: segmentInfo.encryptedBytes.byteLength
```

Comment:
I think it might be time for a helper function to make these objects. Something so that we can do:

```
  ...
  encrypted: transferTypedArray(segmentInfo.encryptedBytes),
  key: transferTypedArray(segmentInfo.key.bytes),
  iv: transferTypedArray(segmentInfo.key.iv)
},
...
```